### ISSUE TITLE
cisstVector Request: Accessors for a subset of fixed-size vector.

### ISSUE THREAD

**Issue by marcinbalicki** [jhu-cisst/cisst#17]
Would be nice to have a clean method to perform assignment (copy) between "subvectors" of fixed-sized vectors. 
```C++
vct5 v5;
vc6 v6;
...
v5(1,3) = v6(2,3);
```
where 
v5(a,b),   a is the start index, b is the length of the subvector.

**Comment by adeguet1** [comment]
I started implementing this on fixed size vectors in branch feature-vector-ref.  Syntax is v.Ref<__size>(startPosition).   See examples of use in:
https://github.com/jhu-cisst/cisst/blob/feature-vector-ref/cisstVector/examples/tutorial/ExampleReferences.h

Output looks like:
```
AsColMatrix:
     0.00000
     1.00000
     2.00000
     3.00000
AsRowMatrix:
     0.00000      1.00000      2.00000      3.00000
Ref<2>(1):
     1.00000      2.00000
Ref<2>(3) on a vector of 4 elements will throw an out of range exception
Exception received: vctFixedSizeVectorRef SetRef out of range
v4, last 2 elements replaced:
     0.00000      1.00000      5.50000      5.50000
v4, first 2 elements replaced:
    -1.00000     -1.00000      5.50000      5.50000
```

All input is welcome.  I plan to implement a similar method on dynamic vectors, syntax will be: v.Ref(size, startPosition).

I'll check if we can assign between fixed size and dynamic Ref too so one could do: vFixed4.Ref<2>(0) = vDynamic.Ref(2, 10); // assuming vDynamic.size() large enough.

**Comment by marcinbalicki** [comment]
Looks great. Should the dynamic version also use the same template syntax? Is it possible to have the same syntax for both?

**Comment by adeguet1** [comment]
Take a look at example file (link above) for features added for dynamic vectors and assigning back and forth between fixed and dynamic Ref (ed42aa5d1587d3a8c72665e1ce61bf2b0e6d3403).   Also found by accident the already existing & operator, quite cool, see line 65 in example!

**Comment by marcinbalicki** [comment]
Looks great.

I recommend to use this example (no ambiguity):
vA.Ref(4) = vB.Ref(6, 4); // assign last 4 vB to first 4 elements of vA

instead of :
vA.Ref(5) = vB.Ref(5, 5); // assign second half of vB to first 5 elements
of vA


Marcin

**Comment by adeguet1** [comment]
Added code and example for Ref() on fixed size and dynamic matrices (see 22b2201303278c6eeb19423829a89e481377bce3).
